client/editor/tool_tile.cpp

```diff
@@ -133,169 +151,173 @@ void editor_tool_tile::update_ett(struct tile *ptile)
-    return shut;
-  } else {
-    return Q_("?hut:None");
-  }
-}
-
-/**
- * \brief Return a string of any bases on a tile
- */
-QString editor_tool_tile::get_tile_base_name(const struct tile *ptile) const
-{
-  QString sbase;
-  extra_type_by_cause_iterate(EC_BASE, pextra)
-  {
-    if (tile_has_extra(ptile, pextra)) {
-      sbase + extra_name_translation(pextra);
+  QVector<QString> names;
+  for (auto cause : causes) {
+    switch (cause) {
+    case EC_ROAD:
+      extra_type_by_cause_iterate(EC_ROAD, pextra)
+      {
+        if (tile_has_extra(ptile, pextra)) {
+          names.push_back(
+              create_help_link(extra_name_translation(pextra), HELP_EXTRA));
+        }
+      }
+      extra_type_by_cause_iterate_end;
+      break;
+    case EC_IRRIGATION:
+      extra_type_by_cause_iterate(EC_IRRIGATION, pextra)
+      {
+        if (tile_has_extra(ptile, pextra)) {
+          names.push_back(
+              create_help_link(extra_name_translation(pextra), HELP_EXTRA));
+        }
+      }
+      extra_type_by_cause_iterate_end;
+      break;
+    case EC_MINE:
+      extra_type_by_cause_iterate(EC_MINE, pextra)
+      {
+        if (tile_has_extra(ptile, pextra)) {
+          names.push_back(
+              create_help_link(extra_name_translation(pextra), HELP_EXTRA));
+        }
+      }
+      extra_type_by_cause_iterate_end;
+      break;
+    case EC_POLLUTION:
+      extra_type_by_cause_iterate(EC_POLLUTION, pextra)
+      {
+        if (tile_has_extra(ptile, pextra)) {
+          names.push_back(
+              create_help_link(extra_name_translation(pextra), HELP_EXTRA));
+        }
+      }
+      extra_type_by_cause_iterate_end;
+      break;
+    case EC_FALLOUT:
+      extra_type_by_cause_iterate(EC_FALLOUT, pextra)
+      {
+        if (tile_has_extra(ptile, pextra)) {
+          names.push_back(
+              create_help_link(extra_name_translation(pextra), HELP_EXTRA));
+        }
+      }
+      extra_type_by_cause_iterate_end;
+      break;
+    case EC_HUT:
+      extra_type_by_cause_iterate(EC_HUT, pextra)
+      {
+        if (tile_has_extra(ptile, pextra)) {
+          names.push_back(
+              create_help_link(extra_name_translation(pextra), HELP_EXTRA));
+        }
+      }
+      extra_type_by_cause_iterate_end;
+      break;
+    case EC_BASE:
+      extra_type_by_cause_iterate(EC_BASE, pextra)
+      {
+        if (tile_has_extra(ptile, pextra)) {
+          names.push_back(
+              create_help_link(extra_name_translation(pextra), HELP_EXTRA));
+        }
+      }
+      extra_type_by_cause_iterate_end;
+      break;
+    case EC_RESOURCE:
+      extra_type_by_cause_iterate(EC_RESOURCE, pextra)
+      {
+        if (tile_has_extra(ptile, pextra)) {
+          names.push_back(
+              create_help_link(extra_name_translation(pextra), HELP_EXTRA));
+        }
+      }
+      extra_type_by_cause_iterate_end;
+      break;
+    case EC_APPEARANCE:
+    case EC_COUNT:
+      break;
     }
```

Comment:
I think this should work:
```suggestion
    extra_type_by_cause_iterate(cause, pextra)
    {
      if (tile_has_extra(ptile, pextra)) {
        names.push_back(
            create_help_link(extra_name_translation(pextra), HELP_EXTRA));
      }
    }
    extra_type_by_cause_iterate_end;
```
Not sure what `EC_APPEARANCE` is for, does it hurt to keep it?

Reply:
Ah ok. I couldn't figure out how to do the full for loop so I did the switch instead.